File: 2015/d01.py
DAY,_,_=__file__.rpartition('.')

from alg.file import download_input
download_input(DAY)

s=''

with open(f'{DAY}.txt','r')as F:
    for l in F:
        l=l.rstrip('\n')
        s+=l

#part 1
print(s.count('(')-s.count(')'))

#part 2
l=0
for i,c in enumerate(s,1):
    if c=='(':l+=1
    else:l-=1
    if l<0:break
print(i)

File: Practice/2015/d01.rb
DAY,_,_ = __FILE__.rpartition(?.)

require'prime'
require'z3'

r=[]
t=''
s=0

File.open("#{DAY}.txt", "r") { |f|
  f.each_line {|l|
    l=l.chomp
    r+=[l]
    t+=l
  }
}

p t.count(?()-t.count(?))

f=0
i=0
t.chars.map{|c|
  f+= c=='(' ? 1 : -1
  i+=1
  f<0 && (p i;exit(0))
}